pandas/lib.pyx

```diff
@@ -997,7 +1001,9 @@ cpdef ndarray[object] astype_str(ndarray arr):
         ndarray[object] result = np.empty(n, dtype=object)
 
     for i in range(n):
```

Comment:
this should simply be checked once for the array (and don't call the unsafe function)